lib/BridgeAbstract.php

```diff
@@ -238,6 +245,52 @@ function($i){ return $i['name']; }, // Just display parameter names
 
 	}
 
+	/**
+	 * Loads configuration for the bridge
+	 *
+	 * Returns errors and aborts execution if the provided configuration is
+	 * invalid.
+	 *
+	 * @return void
+	 */
+	public function loadBridgeConfiguration() {
+		foreach(static::CONFIGURATION as $optionName => $optionValue) {
+
+			$configurationOption = Configuration::getConfig(get_class($this), $optionName);
+
+			if($configurationOption !== null) {
+				$this->configuration[$optionName] = $configurationOption;
+				continue;
+			}
+
+			if($optionValue['required']) {
+				returnClientError(
+					'Missing configuration option :'
+					. $optionName
+				);
+			} else {
+				$this->configuration[$optionName] = $optionValue['defaultValue'];
+			}
+
+		}
+	}
+
+
+	/**
+	 * Check if configuration allows for this bridge loading
+	 *
+	 * @return boolean
+	 */
+	public function isConfigurationValid() {
+		foreach(static::CONFIGURATION as $optionName => $optionValue) {
+			if($optionValue["required"] && Configuration::getConfig(get_class($this), $optionName) === null) {
```

Comment:
Same as before, `required` may not be defined.

```suggestion
			if(isset($optionValue['required']) && $optionValue['required'] === true && Configuration::getConfig(get_class($this), $optionName) === null) {
```